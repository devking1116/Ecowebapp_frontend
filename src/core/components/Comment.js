/* @flow */
import React from 'react';
import reactCSS from 'reactcss';
import { Feed, Icon, Label, Menu, Select, Segment, Dropdown } from 'semantic-ui-react';

type Props = {
  author: string,
  date: string,
  body: string,
  status?: 'active' | 'in-active',
  actionsTakens: Array<string>,
  children?: any,
};

/**
 * The component that displays a comment.
 */
export default function Comment(props: Props): React.Element<*> {
  const { author, date, body, status, actionsTakens, children, ...otherProps } = props;

  const styles = reactCSS({
    'default': {
      actionMenu: {
        display: 'flex',
        justifyContent: 'flex-end',
      },
      feedContainer: {
        position: 'relative',
      },
      content: {
        marginTop: -15,
      },
      feedIcon: {
        fontSize: 10,
        marginTop: 20,
        marginLeft: -20,
        position: 'absolute',
      },
      summary: {
        width: 200,
        display: 'table-cell',
        borderRight: '1px solid #d4d4d5',
        fontSize: 14,
      },
      status: {
        width: 100,
        display: 'table-cell',
      },
      select: {
        fontSize: 14,
        border: 'none',
        paddingLeft: 42,
      },
      selectIcon: {
        position: 'absolute',
        zIndex: '1001',
        fontSize: 14,
        paddingLeft: 18,
        paddingTop: 10,
      },
      fullWidth: {
        width: '100%',
      },
      dropdown: {
        float: 'right',
        fontSize: 20,
        paddingRight: 15,
        marginTop: -30,
      },
      blueText: {
        color: '#4183c4',
      },
      redText: {
        color: 'red',
      },
      actionsTaken: {
        display: 'block',
        fontSize: 14,
      },
      actionsBox: {
        border: '1px solid #d4d4d5',
        paddingLeft: 2,
        paddingRight: 2,
        fontSize: 14,
        marginLeft: 10,
      },
      logBook: {
        marginTop: 5,
      },
    },
  });

  return (
    <Feed.Event>
      <Feed.Label style={styles.feedContainer}>
        <img src='http://semantic-ui.com/images/avatar/small/elliot.jpg' />
      {status === 'active'?
        <Icon name="circle" color="green" style={styles.feedIcon} />:
        null
      }
      {status === 'in-active'?
        <Icon name="circle" color="red" style={styles.feedIcon} />:
        null
      }
      </Feed.Label>
      <Feed.Content style={styles.content}>
        <Feed.Meta style={styles.fullWidth}>
          <div style={styles.summary}>
            <span style={styles.blueText}>{author}</span><br/>
            <span>{date}</span>
          </div>
          <div style={styles.status}>
            <Icon name="power" color={(status !=='active' && status !== 'in-active')? "grey": status == 'active'? "green": "red"} style={styles.selectIcon}/>
            <Select
              placeholder="Status"
              value={status? status : "none" }
              icon={null}
              style={styles.select}
              options={[
                { value: 'active', icon: { name: 'power', color: 'green'}, text: 'Active' },
                { value: 'in-active',icon: { name: 'power', color: 'red'}, text: 'In-Active' },
                { value: 'none', icon: { name: 'power', color: 'grey'}, text: 'None' },
              ]}
            />
          </div>
          <Dropdown pointing icon='angle down' style={styles.dropdown} className='link item top right'>
            <Dropdown.Menu>
              <Dropdown.Item style={styles.blueText}>Edit</Dropdown.Item>
              <Dropdown.Item style={styles.blueText}>Link</Dropdown.Item>
              <Dropdown.Item style={styles.blueText}>Share</Dropdown.Item>
              <Dropdown.Item style={styles.blueText}>Add to Logbooks</Dropdown.Item>
              <Dropdown.Divider />
              <Dropdown.Item style={styles.redText}>Delete</Dropdown.Item>
            </Dropdown.Menu>
          </Dropdown>
        </Feed.Meta>
        <Feed.Meta style={styles.actionsTaken}>
          <span>Actions Taken</span>
          {actionsTakens ? actionsTakens.map((actionstaken, index) =>
            <span style={styles.actionsBox} key={index}>{actionstaken}</span>
          ): null}
        </Feed.Meta>
        <Feed.Extra text>
          {body}
        </Feed.Extra>
        <Feed.Meta style={styles.logBook}>
          {children}
        </Feed.Meta>
      </Feed.Content>
    </Feed.Event>
  );
}
