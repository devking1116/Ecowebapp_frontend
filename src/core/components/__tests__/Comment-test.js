import React from 'react';
import renderer from 'react-test-renderer';

import Comment from '../Comment';

const testComment = {
  author: 'User 0',
  date: '1/1/2016',
  body: 'Test comment 0',
  status: 'active',
  actionsTakens: ['Spoke to Resident', 'Jetted Sewer Line'],
  linkedLogbook: 'logbook 0',
};

test('it displays the given comment.', () => {
  const component = renderer.create(
    <Comment {...testComment} />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
