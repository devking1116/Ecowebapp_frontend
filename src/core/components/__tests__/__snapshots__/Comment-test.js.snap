exports[`test it displays the given comment. 1`] = `
<div
  className="event">
  <div
    className="label"
    style={
      Object {
        "position": "relative",
      }
    }>
    <img
      src="http://semantic-ui.com/images/avatar/small/elliot.jpg" />
    <i
      aria-hidden="true"
      className="green circle icon"
      style={
        Object {
          "fontSize": 10,
          "marginLeft": -20,
          "marginTop": 20,
          "position": "absolute",
        }
      } />
  </div>
  <div
    className="content"
    style={
      Object {
        "marginTop": -15,
      }
    }>
    <div
      className="meta"
      style={
        Object {
          "width": "100%",
        }
      }>
      <div
        style={
          Object {
            "borderRight": "1px solid #d4d4d5",
            "display": "table-cell",
            "fontSize": 14,
            "width": 200,
          }
        }>
        <span
          style={
            Object {
              "color": "#4183c4",
            }
          }>
          User 0
        </span>
        <br />
        <span>
          1/1/2016
        </span>
      </div>
      <div
        style={
          Object {
            "display": "table-cell",
            "width": 100,
          }
        }>
        <i
          aria-hidden="true"
          className="green power icon"
          style={
            Object {
              "fontSize": 14,
              "paddingLeft": 18,
              "paddingTop": 10,
              "position": "absolute",
              "zIndex": "1001",
            }
          } />
        <div
          aria-busy={undefined}
          aria-disabled={undefined}
          aria-expanded={false}
          aria-multiselectable={undefined}
          className="ui selection dropdown"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          role="listbox"
          style={
            Object {
              "border": "none",
              "fontSize": 14,
              "paddingLeft": 42,
            }
          }
          tabIndex={0}>
          <select
            aria-hidden="true"
            multiple={undefined}
            name={undefined}
            type="hidden"
            value="active">
            <option
              value="" />
            <option
              value="active">
              Active
            </option>
            <option
              value="in-active">
              In-Active
            </option>
            <option
              value="none">
              None
            </option>
          </select>
          <div
            className="text">
            Active
          </div>
          <div
            className="menu transition">
            <div
              aria-checked={true}
              aria-disabled={undefined}
              aria-selected={true}
              className="active selected item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }>
              <i
                aria-hidden="true"
                className="green power icon" />
              Active
            </div>
            <div
              aria-checked={false}
              aria-disabled={undefined}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }>
              <i
                aria-hidden="true"
                className="red power icon" />
              In-Active
            </div>
            <div
              aria-checked={false}
              aria-disabled={undefined}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }>
              <i
                aria-hidden="true"
                className="grey power icon" />
              None
            </div>
          </div>
        </div>
      </div>
      <div
        aria-busy={undefined}
        aria-disabled={undefined}
        aria-expanded={false}
        aria-multiselectable={undefined}
        className="ui pointing link item top right dropdown"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        role="listbox"
        style={
          Object {
            "float": "right",
            "fontSize": 20,
            "marginTop": -30,
            "paddingRight": 15,
          }
        }
        tabIndex={0}>
        <div
          className="text" />
        <i
          aria-hidden="true"
          className="angle down icon" />
        <div
          className="menu transition">
          <div
            aria-checked={undefined}
            aria-disabled={undefined}
            aria-selected={undefined}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "color": "#4183c4",
              }
            }>
            Edit
          </div>
          <div
            aria-checked={undefined}
            aria-disabled={undefined}
            aria-selected={undefined}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "color": "#4183c4",
              }
            }>
            Link
          </div>
          <div
            aria-checked={undefined}
            aria-disabled={undefined}
            aria-selected={undefined}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "color": "#4183c4",
              }
            }>
            Share
          </div>
          <div
            aria-checked={undefined}
            aria-disabled={undefined}
            aria-selected={undefined}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "color": "#4183c4",
              }
            }>
            Add to Logbooks
          </div>
          <div
            className="divider" />
          <div
            aria-checked={undefined}
            aria-disabled={undefined}
            aria-selected={undefined}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "color": "red",
              }
            }>
            Delete
          </div>
        </div>
      </div>
    </div>
    <div
      className="meta"
      style={
        Object {
          "display": "block",
          "fontSize": 14,
        }
      }>
      <span>
        Actions Taken
      </span>
      <span
        style={
          Object {
            "border": "1px solid #d4d4d5",
            "fontSize": 14,
            "marginLeft": 10,
            "paddingLeft": 2,
            "paddingRight": 2,
          }
        }>
        Spoke to Resident
      </span>
      <span
        style={
          Object {
            "border": "1px solid #d4d4d5",
            "fontSize": 14,
            "marginLeft": 10,
            "paddingLeft": 2,
            "paddingRight": 2,
          }
        }>
        Jetted Sewer Line
      </span>
    </div>
    <div
      className="text extra">
      Test comment 0
    </div>
    <div
      className="meta"
      style={
        Object {
          "marginTop": 5,
        }
      } />
  </div>
</div>
`;
